tensorflow/core/common_runtime/mkl_threadpool_device_test.cc

```diff
@@ -0,0 +1,62 @@
+/* Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#ifdef INTEL_MKL
+
+#include "tensorflow/core/common_runtime/threadpool_device.h"
+
+#include "tensorflow/core/lib/core/status_test_util.h"
+#include "tensorflow/core/platform/cpu_info.h"
+#include "tensorflow/core/platform/logging.h"
+#include "tensorflow/core/platform/test.h"
+#include "tensorflow/core/public/session_options.h"
+
+namespace tensorflow {
+
+#ifdef _OPENMP
+TEST(MKLThreadPoolDeviceTest, TestOmpDefaults) {
+  SessionOptions options;
+  unsetenv("OMP_NUM_THREADS");
```

Comment:
Please get rid of anything manipulating the environment.

Reply:
There are other places in Tensorflow unit tests where unsetenv is being used to ensure we are checking the default paths (https://github.com/tensorflow/tensorflow/blob/a3d90a967e3c1c8d93928ca6620bbfa2fc8ce3a1/tensorflow/core/platform/cloud/google_auth_provider_test.cc#L71).

Do you have an alternative that I could look into?

Reply:
I can't find any thread-safe versions of these APIs. Maybe we can put locks around the calls since these are just tests?

Reply:
I think it is OK in unit tests.